modules/oe_subscriptions_anonymous/tests/src/Functional/HtmlMailsTest.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\oe_subscriptions_anonymous\Functional;
+
+use Drupal\Core\Url;
+use Drupal\symfony_mailer\Email;
+use Drupal\symfony_mailer_test\MailerTestTrait;
+use Drupal\Tests\BrowserTestBase;
+use Drupal\Tests\flag\Traits\FlagCreateTrait;
+use Drupal\Tests\oe_subscriptions_anonymous\Trait\StatusMessageTrait;
+use Symfony\Component\DomCrawler\Crawler;
+
+/**
+ * Tests the HTML in mails.
+ */
+class HtmlMailsTest extends BrowserTestBase {
```

Comment:
I think the idea behind the test is not right at the moment, it doenst actually check html and most of what it tests is already tested in other tests. Once the refactoring for optional html email is done, then I think we can rethink the existing traits to account for this, and change the anonymous testing or even add a new test if needed.

Reply:
Lets try to get the hash similar to what was done in SubscriptionRequestRoutesTest, give it a try

Reply:
I've adapted the test, now we have one functions to check each mails and I check that the html is present and where, about links with tokens couldn't use the manager to get the URL, the service generates a new token every time you resquest it. I check the link performing the click instead, I think that is better than before.